olive/passes/olive_pass.py

```diff
@@ -211,14 +215,52 @@ def run(
                 )
                 components.append(output_model_component)
                 component_names.append(component_name)
+                Pass._carry_forward_additional_files(component_model, output_model_component)
             output_model = CompositeModelHandler(components, component_names)
         else:
             output_model = self._run_for_config(model, data_root, config, output_model_path)
+            Pass._carry_forward_additional_files(model, output_model)
+
         # assumption: the model attributes from passes, if any, are more important than
         # the input model attributes, we should not update/extend anymore outside of the pass run
         output_model.model_attributes = output_model.model_attributes or model.model_attributes
         return output_model
 
+    @staticmethod
+    def _carry_forward_additional_files(input_model: OliveModelHandler, output_model: OliveModelHandler):
+        # NOTE: Can't use model.model_path because that always gets resolved to a filepath.
+        # We need the directory path here.
+        input_model_output_path = input_model.get_resource("model_path")
+        if not input_model_output_path:
+            return
+
+        input_model_output_path = Path(input_model_output_path)
+        if not input_model_output_path.is_dir():
+            return
+
+        input_model_attributes = input_model.model_attributes or {}
+        input_model_additional_files = set(input_model_attributes.get("additional_files", []))
+        if not input_model_additional_files:
+            return
+
+        output_model_output_path = Path(output_model.get_resource("model_path"))
+        if not output_model_output_path.is_dir():
+            raise ValueError(f"Expecting {output_model_output_path} to be a directory.")
+
+        output_model_attributes = output_model.model_attributes or {}
+        output_model_additional_files = set(output_model_attributes.get("additional_files", []))
+
+        for filename in input_model_additional_files:
+            output_model_additional_files.add(filename)
+            output_filepath = output_model_output_path / filename
+
+            if not output_filepath.exists():
+                input_filepath = input_model_output_path / filename
+                shutil.copy(input_filepath, output_filepath)
```

Comment:
Can you add a TODO here to investigate using symlink here?

Reply:
Use of symlink is a wider issue not specific to this use case only. I think the issue deserves to be on our work item board. Will add a comment here as well.